docs/quickstart.mdx

```diff
@@ -90,11 +90,163 @@ This is a Sapphire tutorial and you're already a Solidity expert, so let's not
 bore you with explaining the gritty details of the contract.
 Start by pasting [Vigil.sol] into `contracts/Vigil.sol`.
 
-While you're there, also place [run-vigil.ts] into `scripts/run-vigil.ts`.
-We'll need that later.
-
-[Vigil.sol]: https://github.com/oasisprotocol/sapphire-paratime/blob/stable/clients/js/1.x/examples/hardhat/contracts/Vigil.sol
-[run-vigil.ts]: https://github.com/oasisprotocol/sapphire-paratime/blob/stable/clients/js/1.x/examples/hardhat/scripts/run-vigil.ts
+1. Create a new file called `Vigil.sol` under `contracts/`:
+2. Paste the following contract into it:
+
+  <details>
+    <summary> Vigil.sol Contract </summary>
+
+    ```solidity title="Vigil.sol" showLineNumbers
+    // SPDX-License-Identifier: MIT
+    pragma solidity ^0.8.9;
+
+    contract Vigil {
+        struct SecretMetadata {
+            address creator;
+            string name;
+            /// @notice How long (in seconds) the secret should remain so past the creator's last update.
+            uint256 longevity;
+        }
+
+        event SecretCreated(
+            address indexed creator,
+            string indexed name,
+            uint256 index
+        );
+        event SecretRevealed(
+            address indexed creator,
+            string indexed name,
+            uint256 index
+        );
+
+        SecretMetadata[] public _metas;
+        bytes[] private _secrets;
+        /// @dev The unix timestamp at which the address was last seen.
+        mapping(address => uint256) public _lastSeen;
+
+        function createSecret(
+            string calldata name,
+            uint256 longevity,
+            bytes calldata secret
+        ) external {
+            _updateLastSeen();
+            _metas.push(
+                SecretMetadata({
+                    creator: msg.sender,
+                    name: name,
+                    longevity: longevity
+                })
+            );
+            _secrets.push(secret);
+            emit SecretCreated(msg.sender, name, _metas.length - 1);
+        }
+
+        /// @notice Reveals the secret at the specified index.
+        function revealSecret(uint256 index) external returns (bytes memory) {
```

Comment:
I wonder why not define it as `view`. We don't need to emit the event IMO. This also simplifies the `revealSecret.staticCall(0)` typescript code simply to `revealSecret(0)`.

Reply:
🤔 We don't mention it, but having the event show up in the Explorer is nice.